python-cell.el

```diff
@@ -118,17 +122,28 @@ the command `python-cell-mode' to turn Python-Cell mode on."
              (forward-char 1))
     (goto-char (point-min))))
 
-(defun python-cell-beginning-of-cell (&optional arg)
-  (interactive "p")
-  ;; TODO: prefix support
+(defun python-cell-beginning-of-matched-cellbreak ()
+  (goto-char (match-beginning 0))
+  (beginning-of-line))
 
+(defun python-cell-end-of-matched-cellbreak ()
+  (goto-char (match-end 0))
+  (end-of-line)
+  (forward-char 1))
+  
+
+(defun python-cell-beginning-of-cell--sub (cellbreak-goto-function)
```

Comment:
I have doubts about the naming of this function and the refactoring I did related to it. Essentially, I refactored the function `python-cell-beginning-of-cell` into more functions to have better control of where at the beginning the cursor is placed:

* When I navigate to a cell, I want the cursor where the actual code starts, that is *after* the cellbreak string (`if False...` or `##....`)
* When I send a cell to the interpreter, I want the cursor to be at the very beginning of the cell, including the cellbreak string.

To accomplish this, the function `python-cell-beginning-of-cell--sub` accepts a function as argument that will put the cursor in the right place in case a cellbreak is detected. Either we pass `python-cell-beginning-of-matched-cellbreak` or `python-cell-end-of-matched-cellbreak` to that function. The original `python-cell-beginning-of-cell` function passes in the former.

Is there a better way to structure this code? What about naming? Feel free to comment on this and I will address it.

Reply:
I wonder if the code would be more clear if I instead had a function something like

```elisp
(defun python-cell--detect-beginning-of-cell ()
  (end-of-line)
    (if (re-search-backward python-cell-cellbreak-regexp nil t)
        nil
      (progn (goto-char (point-min)) nil)))
```
that returns a truthy value in case the beginning of the cell is detected. And then, in that case, the caller of that function can move the cursor:
```elisp
(when (python-cell--detect-beginning-of-cell)
   (python-cell-end-of-matched-cellbreak))
```

This might make the code more readable but I am really not sure.

Reply:
> * When I send a cell to the interpreter, I want the cursor to be at the very beginning of the cell, including the cellbreak string.

So you want to send the `if False` string to the python shell s.t. the body of the `if False` statement is not executed by the python shell? Why that?

Reply:
Because in the function `python-cell-shell-send-cell` we call `python-cell-preprocess-cell` that will replace `if False:` by `if True:`. Because of this replacement, the body of the cell will be run.

For instance, if we have a cell

```py
if False: # load the dataset
    ds = load_dataset("data.csv")
```

the string sent to Python becomes

```py
if True: # load the dataset
    ds = load_dataset("data.csv")
```

Is there a better way to accomplish the effect of having cells with `if False:` executed?

Reply:
Yes, detecting the end of rich-comments would mean that the following code

```py
if False:
    dataset = load_dataset("validation.py")

results = algorithm.evaluate(dataset);
```

would actually become two cells, right? The two cells being

```py
if False:
    dataset = load_dataset("validation.py")
```

and

```py

results = algorithm.evaluate(dataset);
```

I am not sure how useful that would be and it would probably be tricky to get it work. So maybe it is something to consider outside of the scope of this PR.